### PR TITLE
Propagate the incoming authentication configuration

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Needs review from @sguilhen

### CODE REVIEW

File: ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java [hunk 1]
```diff
@@ -307,21 +311,28 @@ public OutputStream _invoke(final String opName, final InputStream in, final Res
...
-                                identity = this.authenticate(principal, (char[]) credential);
+                                authenticationConfiguration = authenticationConfiguration.useName(principal.getName())
+                                        .usePassword(password);
+                                identity = this.authenticate(principal, password);
                             }
                             final InterceptorContext interceptorContext = new InterceptorContext();
                             this.prepareInterceptorContext(op, params, interceptorContext);
                             try {
-                                retVal = identity.runAs((PrivilegedExceptionAction<Object>) () -> this.componentView.invoke(interceptorContext));
+                                final AuthenticationContext context = AuthenticationContext.empty().with(MatchRule.ALL, authenticationConfiguration);
```
Comment (dmlloyd) [inline ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java]: This seems odd.  Where is this auth info reused?  On subsequent outbound IIOP invocations?

Comment (stuartwdouglas) [reply ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java]: Yes, some of the test seem to expect the identity to be propagated from appclient -> Wildfly -> Glassfish

Comment (dmlloyd) [reply ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java]: I wonder if we should be defaulting to forwarding authentication information.  @darranl what do you think?

I know we definitely don't want a blanket context at any rate.

Comment (darranl) [reply ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java]: We are one step closer to being able to support it.

For forward authentication to work the security domain needs to be available to the AuthenticationConfiguration, however now it is associated with the deployments class loader we could have a default mode that uses the security domsin from the deployment.

Comment (dmlloyd) [reply ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java]: OK for now I think what we can do is, change AuthenticationContext.empty() to captureCurrent(), and then make the match rule only match protocol "iiop", and that will effectively forward the incoming authentication.

Comment (dmlloyd) [reply ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java]: @sguilhen CC / I think Stuart's change is 98% workable, we just need this last adjustment.

Comment (dmlloyd) [reply ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java]: (we can add "proper" forwarding later)

**Pull request merged**